### PR TITLE
Java - How Range Operations Work

### PR DESCRIPTION
**Pull request by kapnobatai137**

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/java/fundamentals/tips-ii/how-range-operations-work.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/java/fundamentals/tips-ii/how-range-operations-work.md?ref=98728b4bd8df29bbce3991c2a68a4c64a82a0405)

File: java/fundamentals/tips-ii/how-range-operations-work.md [hunk 1]
```diff
@@ -1,46 +1,55 @@
...
 ## Content
 
-When specifying a range of values, the first value `from` is included while the last value `to` is excluded.
+You can use the `substring()` method to extract a part of a string, **as a new string**.
+
+The syntax is:
 
 ```java
-String.substring(from,to); 
+String substring(
```
Comment (Loopiezlol) [inline java/fundamentals/tips-ii/how-range-operations-work.md:25]:
```suggestion
String.substring(
```

File: java/fundamentals/tips-ii/how-range-operations-work.md [hunk 2]
```diff
@@ -1,46 +1,55 @@
...
 ---
 
-# How range operations work
+# How Range Operations Work
 
 ---
 ## Content
 
-When specifying a range of values, the first value `from` is included while the last value `to` is excluded.
+You can use the `substring()` method to extract a part of a string, **as a new string**.
```
Comment (Loopiezlol) [inline java/fundamentals/tips-ii/how-range-operations-work.md:20]:
```suggestion
Strings can be thought of as a list of characters. Hence, a substring is essentially a range in the original string. 
You can use the `substring()` method to extract a part of a string. The result will be returned **as a new value** while the original string isn't mutated. 
```

File: java/fundamentals/tips-ii/how-range-operations-work.md [hunk 3]
```diff
@@ -1,46 +1,55 @@
...
 
 notes: 'https://insights.enki.com/#/contrib/others/55c83ba83f06ab320051aa8a?search=kha'
 
 ---
 
-# How range operations work
+# How Range Operations Work
 
 ---
 ## Content
```
Comment (Loopiezlol) [inline java/fundamentals/tips-ii/how-range-operations-work.md:19]: Can we add a very short intro paragraph about what a range is in this context? Ranges can mean different things, but here we mean sequences of values bounded by position

"A range can refer to an arbitrary, ordered sequence of values within a list-like value. The boundaries of a range are often represented by indexes."

File: java/fundamentals/tips-ii/how-range-operations-work.md [hunk 4]
```diff
@@ -1,46 +1,55 @@
...
-List.sublist(from,to);
+List.sublist(
+  int startIndex,
+  int endIndex
+);
 ```
 
 ```java
-Array.copyOfRange(T[],from,to);
+Array.copyOfRange(
```
Comment (Loopiezlol) [inline java/fundamentals/tips-ii/how-range-operations-work.md:48]: this is very confusing I have no idea what it refers to 🤣

instead, we can remove it and show a concrete example of lists. define a list with 4 values and retrieve one of its sublists

Comment (kapnobatai137) [reply java/fundamentals/tips-ii/how-range-operations-work.md:49]: something like this:

```java
List<Integer> myList = new ArrayList<Integer>();
for (int i = 0; i <= 4; i++) {
  myList.add(i);
}
List<Integer> newList = myList.subList(0, 3);
System.out.println(newList);
```

or something like this

```java
myList = [0, 1, ,2, 3, 4]
mySublist = myList.subList(0, 3);
// [0, 1, 2]
```

Comment (Loopiezlol) [reply java/fundamentals/tips-ii/how-range-operations-work.md:51]: I think the first example yeah - it points out how `.sublist` works based on ranges

**Pull request merged**